### PR TITLE
Update CMakeLists.txt

### PR DESCRIPTION
**Pull request by compnerd**
Update the CMakeLists.txt to contain the latest changes. This also introduces a new option `SWIFT_FIREBASE_BUILD_EXAMPLES` to allow disabling the examples. This change is motivated by the desire to cross-compile this to other platforms (e.g. Android), which is not possible via SPM.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,20 +30,25 @@ target_link_directories(firebase INTERFACE
...
+  firebase)
 target_link_libraries(FirebaseCore PRIVATE
   libcurl
-  firebase
+  firebase_app
+  flatbuffers
   zlibstatic)
 
-add_library(FirebaseAuth
+add_library(FirebaseAuth SHARED
```
Comment (fiedukow) [inline CMakeLists.txt:50]: `Sources/FirebaseAuth/FirebaseEmailAuthProvider.swift` seem to be missing?

Comment (compnerd) [reply CMakeLists.txt:50]: Ugh, it does indeed. Thanks!

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,8 +11,16 @@ set(CMAKE_Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift)
...
+option(SWIFT_FIREBASE_BUILD_EXAMPLES "Build example UI" TRUE)
+
+# ${TOOLCHAIN_ROOT}/usr/include allows us to access swift C++ interop headers.
+cmake_path(GET CMAKE_Swift_COMPILER PARENT_PATH _SWIFT_INCLUDE_DIR)
+cmake_path(GET _SWIFT_INCLUDE_DIR PARENT_PATH _SWIFT_INCLUDE_DIR)
+include_directories(SYSTEM ${_SWIFT_INCLUDE_DIR}/include)
+
 add_library(firebase INTERFACE)
 target_compile_options(firebase INTERFACE
+  "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xcc -DSR69711>"
```
Comment (fiedukow) [inline CMakeLists.txt:23]: It might be just me, but does `SR69711` warrant some comment? It's kinda cryptic - at least for me, and even google search doesn't really give much information about it.

File: CMakeLists.txt [hunk 3]
```diff
@@ -22,46 +30,74 @@ target_link_directories(firebase INTERFACE
   third_party/firebase-development/usr/libs/windows/deps/app
   third_party/firebase-development/usr/libs/windows/deps/app/external)
 
-add_library(FirebaseCore
+add_library(FirebaseCore SHARED
```
Comment (fiedukow) [inline CMakeLists.txt:33]: Is this change intentional?

Comment (compnerd) [reply CMakeLists.txt:33]: Yes, this change was intentional (IIRC, we ended up making them dynamically linked in Arc as well).

**Review by fiedukow — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (compnerd) [reply CMakeLists.txt:23]: I don't think it really does - this is basically the same as WPP-????. It is a bug report identifier (https://github.com/apple/swift/issues/69711).

**Review by darinf — COMMENTED** [review]
LGTM
